src/Uno.UI/Microsoft/UI/Xaml/Controls/TreeView/TreeViewItem.cs

```diff
@@ -0,0 +1,778 @@
+
+				if (childIndex != 0)
+				{
+					if (targetNode.IsExpanded)
+					{
+						targetNode.IsExpanded = false;
+					}
+
+					ReorderItems(listControl, targetNode, position, childIndex, false);
+				}
+			}
+			else if ((key == VirtualKey.Down || key == VirtualKey.Right) && (position != listControl.Items.Count - 1))
+			{
+				var childIndex = parentNode.Children.IndexOf(targetNode);
+				if (childIndex != parentNode.Children.Count - 1)
+				{
+					if (targetNode.IsExpanded)
+					{
+						targetNode.IsExpanded = false;
+					}
+
+					ReorderItems(listControl, targetNode, position, childIndex, true);
+				}
+			}
+		}
+
+		private bool HandleExpandCollapse(VirtualKey key)
+		{
+			var treeView = AncestorTreeView;
+			var targetNode = treeView.NodeFromContainer(this);
+			var flowDirectionReversed = FlowDirection == FlowDirection.RightToLeft;
+			bool isExpanded = targetNode.IsExpanded;
+			bool handled = false;
+
+			// Inputs for Collapse/Move to parent
+			if ((key == VirtualKey.Left && !flowDirectionReversed) ||
+				(key == VirtualKey.Right && flowDirectionReversed))
+			{
+				if (isExpanded)// Is expanded : need to collapse
+				{
+					targetNode.IsExpanded = false;
+					var targetValue = treeView.ContainerFromNode(targetNode);
+					if (targetValue != null)
+					{
+						var targetItem = (TreeViewItem)targetValue;
+						targetItem.Focus(FocusState.Keyboard);
+					}
+
+					handled = true;
+				}
+				else // Is collapsed: need to select parent
+				{
+					var parentNode = targetNode.Parent;
+					if (parentNode != null)
+					{
+						var parentValue = treeView.ContainerFromNode(parentNode);
+						if (parentValue != null)
+						{
+							var parentItem = (TreeViewItem)parentValue;
+							parentItem.Focus(FocusState.Keyboard);
+							handled = true;
+						}
+					}
+				}
+			}
+			// Inputs for Expand/Move to first child
+			else if ((key == VirtualKey.Right && !flowDirectionReversed) ||
+					 (key == VirtualKey.Left && flowDirectionReversed))
+			{
+				if (!isExpanded && targetNode.HasChildren) // Is collapsed : need to expand
+				{
+					targetNode.IsExpanded = true;
+					handled = true;
+				}
+				else if (targetNode.Children.Count > 0) // Is expanded and has children: need to select first child (if applicable)
+				{
+					var childNode = targetNode.Children[0];
+					var childValue = treeView.ContainerFromNode(childNode);
+					if (childValue != null)
+					{
+						var childItem = (TreeViewItem)childValue;
+						childItem.Focus(FocusState.Keyboard);
+						handled = true;
+					}
+				}
+			}
+
+			return handled;
+		}
+
+		private void OnExpandCollapseChevronPointerPressed(object sender, PointerRoutedEventArgs args)
+		{
+			TreeViewNode targetNode = TreeNode;
+			var isExpanded = !targetNode.IsExpanded;
+			targetNode.IsExpanded = isExpanded;
+			args.Handled = true;
+		}
+
+		private void RecycleEvents(bool useSafeGet = false)
+		{
+			//var chevron = m_expandCollapseChevron;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)